### PR TITLE
Replace markdown table

### PR DESCRIPTION
**Pull request by willmcgugan**
Replaces  DataTables in Markdown with Rich tables. It makes more sense to use Rich tables in a document, so that it they will wrap nicely.

Fixes https://github.com/Textualize/textual/issues/1911

## Before

Screenshot 2023-03-14 at 13 30 15

## After

Screenshot 2023-03-14 at 13 30 00

### CODE REVIEW

File: src/textual/widgets/_markdown.py [hunk 1]
```diff
@@ -316,17 +318,59 @@ def compose(self) -> ComposeResult:
...
+        height: auto;
+
+    }
+    MarkdownTableContent > .markdown-table--header {
+        text-style: bold;
+
+    }
+    MarkdownTableContent > .markdown-table--lines {
+
+    }
```
Comment (davep) [inline src/textual/widgets/_markdown.py:334]: Are empty `DEFAULT_CSS` rules needed to make component styles work?

File: src/textual/widgets/_markdown.py [hunk 2]
```diff
@@ -316,17 +318,59 @@ def compose(self) -> ComposeResult:
         self._blocks.clear()
 
 
+class MarkdownTableContent(Widget):
```
Comment (davep) [inline src/textual/widgets/_markdown.py:321]: Class could use a docstring.

File: src/textual/widgets/_markdown.py [hunk 3]
```diff
@@ -316,17 +318,59 @@ def compose(self) -> ComposeResult:
...
+
+    }
+    MarkdownTableContent > .markdown-table--lines {
+
+    }
+    """
+
+    COMPONENT_CLASSES = {"markdown-table--header", "markdown-table--lines"}
+
+    def __init__(self, headers: list[Text], rows: list[list[Text]]):
```
Comment (davep) [inline src/textual/widgets/_markdown.py:339]: The public properties could use docstrings.

**Review by davep — APPROVED** [review]
A question about the CSS and a couple of docstring reminders but looks good otherwise!

**Review by mrossinek — COMMENTED** [review]
Looking at that screenshot in the PR description, I think the text in `example.md` should be updated to not refer to `DataTable` any longer.
That text there also refers to clickable column headers to allow sorting. Not sure if you deem this out of scope now (since this could have been possible based on `DataTable` but not based on `rich.Table`). Or you could update it to say that this is totally if one wants to implement such a feature in a document viewer but with the caveats of wrapping which this PR is solving.

Just some thoughts... :slightly_smiling_face:

File: src/textual/widgets/_markdown.py (hunk 1 above)
Comment (willmcgugan) [reply src/textual/widgets/_markdown.py:334]: No, just a side-effect of editing. Will remove.

**Comment by willmcgugan** [comment]
@mrossinek Sorting by columns should still be possible. It's straightforward to turn those headers in to links.

**Pull request merged**